packages/core/theme-module/theme/pages/Product.vue

```diff
@@ -256,7 +256,8 @@ export default {
       updateFilter,
       configuration,
       product,
-      relatedProducts,
+      // TODO: Temporary. Product search function returns products with all available variants as a flat array. Because of that it omits query limitation.
+      relatedProducts: relatedProducts.value.filter((relatedProduct, i) => i < 8),
```

Comment:
`getFiltered` with `master` set to true - that solution won't work, you have there a flattened array of vartians from multiple products,  so if you cut this starting from the beginning you basically will see just first product from the API (depending on how many variant is has)

Reply:
and of course, create a computed property for  this